rma/models/stock_move.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    # RMAs that were created from the delivery move
+    rma_ids = fields.One2many(
+        comodel_name='rma',
+        inverse_name='move_id',
+        string='RMAs',
+        copy=False,
+    )
+    # RMAs linked to the incoming movement from client
+    rma_receiver_ids = fields.One2many(
+        comodel_name='rma',
+        inverse_name='reception_move_id',
+        string='RMA receivers',
+        copy=False,
+    )
+    # RMA that create the delivery movement to the customer
+    rma_id = fields.Many2one(
+        comodel_name='rma',
+        string='RMA return',
+        copy=False,
+    )
+
+    def unlink(self):
+        if self.mapped('rma_receiver_ids'):
+            raise ValidationError(
+                _("There is at least one move you are trying to "
+                  "delete that is a reception of one or more RMAs."))
+        return super().unlink()
+
+    def _action_done(self):
+        """ Avoids to validate stock.move with less quantity than the
+        quantity in the linked receiver RMA. It also set the appropriated
+        linked RMA to 'received' or 'delivered'.
+        """
+        for move in self.filtered(
+                lambda r: r.state not in ('done', 'cancel')):
+            rma_receiver = move.rma_receiver_ids
+            if (rma_receiver
+                    and move.quantity_done != rma_receiver.product_uom_qty):
+                raise ValidationError(
+                    _("The quantity done for the product '%s' must "
+                      "be equal to its initial demand because the "
+                      "stock move is linked to an RMA (%s).")
```

Comment:
Maybe at this stage we should be able to change the rma product quantity to fit the incoming quantity.

Other (maybe a little more complicated option) would be split the RMA along with the backorder.

Reply:
No, if the quantity doesn't fit, remove the RMA and create another one after the picking backorder split.

Reply:
Ok, maybe a little bit rigid, but not a big deal